addons/mail/static/tests/tours/activity_date_format_tour.js

```diff
@@ -0,0 +1,43 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+
+registry.category("web_tour.tours").add("mail_activity_date_format", {
+    test: true,
+    steps: () => [
+        {
+            trigger: "button:contains('Activities')",
+        },
+        {
+            trigger: "input[id*='activity_type_id']",
+            run: "click",
+        },
+        {
+            trigger: ".dropdown-item:contains('To-Do')",
+        },
+        {
+            trigger: "div[name='summary'] input",
+            run: "text Go Party",
+        },
+        {
+            trigger: "button:contains('Schedule')",
+        },
+        {
+            trigger: ".o-mail-Activity:contains('Go Party')",
+        },
+        {
+            trigger: ".fa-info-circle",
+            run: "click",
+        },
+        // Format expected from the server for the first second of 2024 is date_format = "%d/%b/%y", time_format = "%I:%M:%S %p".
+        {
+            trigger: ".o-mail-Activity-details tr:contains('Created') td:contains('01/Jan/24 12:00:00 AM')",
```

Comment:
`12:00 AM` is quite confusing for many people, and is often mis-used (like AM/PM resets at 12 and not 1, strange isn't it?), so generally people prefer to explicitly say "midnight" (and "noon/midday").

Another argument against using this in test: this makes maintaining the test harder due to confusion.